test/dataset/test_rolling.py

```diff
@@ -0,0 +1,306 @@
+            [10 - i for i in range(8)],  # test 7
+            [10 - i for i in range(8)],  # test 8
+            [10 - i for i in range(8)],  # test 9
+            [3],  # test 10
+        ]
+
+    return convert_to_expected(lengths)
+
+
+def generate_expected_rolled_dataset(pl, unique_rolls, ds_name, ignore_end):
+    to_compare = None
+
+    if ignore_end:
+        to_compare = generate_expected_dataset_varying_open_end(pl)
+    elif unique_rolls and ds_name == "constant":
+        to_compare = generate_expected_dataset_unique(pl)
+    elif ds_name == "constant":
+        to_compare = generate_expected_dataset_standard(pl)
+    elif unique_rolls and ds_name == "varying":
+        to_compare = generate_expected_dataset_varying_unique(pl)
+    elif ds_name == "varying":
+        to_compare = generate_expected_dataset_varying_standard(pl)
+
+    assert to_compare, "dataset to compare with is not implemented"
+
+    return to_compare
+
+
+def generate_dataset(name):
+    dataset = None
+    if name == "constant":
+        _, _, dataset = constant_dataset()
+    elif name == "varying":
+        # Tests edge cases
+        # t0: start time of target
+        # ts: start time of desired range
+        # te: end time of desired range
+        # t1: end time of target
+        # ts < te, t0 <= t1
+        #
+        # start time index of rolling window is 20
+        # end time index of rolling window is 24
+        # ts = 2000-01-01 20:00:00
+        # te = 2000-01-02 00:00:00
+
+        ds_list = [
+            {  # test 1: ends after end time, te > t1
+                "target": [0.0 for i in range(30)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 2: ends at the end time, te == t1
+                "target": [0.0 for i in range(25)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 3: ends between start and end times, ts < t1 < te
+                "target": [0.0 for i in range(23)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 4: ends on start time, ts == t1
+                "target": [0.0 for i in range(20)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 5: ends before start time, t1 < ts
+                "target": [0.0 for i in range(15)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 6: starts on start ends after end, ts == t0, te > t1
+                "target": [0.0 for i in range(10)],
+                "start": pd.Timestamp(2000, 1, 1, 20, 0),
+            },
+            {  # test 7: starts in between ts and te, ts < t0 < te < t1
+                "target": [0.0 for i in range(10)],
+                "start": pd.Timestamp(2000, 1, 1, 22, 0),
+            },
+            {  # test 8: starts on end time, te == t0
+                "target": [0.0 for i in range(10)],
```

Comment:
You can abbreviate these with

```suggestion
                "target": [0.0] *  10],
```